### PR TITLE
Share artifacts between jobs

### PR DESCRIPTION
**Pull request by jprendes**
This PR shares the artifacts created during the build jobs with the e2e test jobs.
It cuts checks times from 20\~25 min to 13\~16 min.

Further improvements can be made by splitting the Ubuntu based jobs and Windows base jobs, since currently Windows builds are in the critical path.

### CODE REVIEW

**Review by Mossaka — DISMISSED** [review]
LGTM! Just had one question

**Comment by Mossaka** [comment]
Also noticed something interesting (not related to this PR):

image


Why are ubuntu 22.04 binaries significantly bigger than these in ubuntu 20.04?

**Comment by jprendes** [comment]
> Why are ubuntu 22.04 binaries significantly bigger than these in ubuntu 20.04?

They are not. WasmEdge binaries are significantly bigger than wasmtime.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -83,12 +84,21 @@ jobs:
...
         run: |
           make test
+      - name: Package artifacts
+        shell: bash
+        run: |
+          make dist RUNTIMES=${{ matrix.runtime }}
+      - name: Upload artifacts
+        uses: actions/upload-artifact@master
+        with:
+          name: containerd-shim-${{ matrix.runtime }}-${{ matrix.os }}
```
[the remark this answers is not in the document]
Comment (jprendes) [reply .github/workflows/ci.yml:100]: They are not. I'm manually resetting the executable permissions. I'm going to follow the advice there and `tar.gz` them before upload.

File: test/k8s/Dockerfile [hunk 2]
```diff
@@ -1,43 +1,22 @@
...
-FROM scratch AS shim
-COPY --from=shim-build /opt/shim/bin/* /
-
-
-# Reuse kind-base so we don't have to download other images...
-FROM kind-base AS kind-fetch
-ARG TARGETARCH
 ARG KIND_VERSION=v0.17.0
-RUN curl -sSLf https://kind.sigs.k8s.io/dl/${KIND_VERSION}/kind-linux-${TARGETARCH} > /root/kind && chmod +x /root/kind
+ARG KIND_NODE_VERSION=v1.23.13
```
Comment (jsturtevant) [inline test/k8s/Dockerfile:4]: can be a follow up but we maybe should consider bumping this to a newer k8s version

**Review by jsturtevant — APPROVED** [review]
lgtm

**Review by Mossaka — APPROVED** [review]

File: test/k8s/Dockerfile (hunk 2 above)
Comment (jprendes) [reply test/k8s/Dockerfile:4]: Created an issue to track this

**Pull request merged**